minio/select_object_reader.py

```diff
@@ -0,0 +1,294 @@
+        # Messages read in chunks of read_buffer bytes
+        chunked_message = self.response.read(read_buffer)
+        total_byte_parsed = 0
+        if len(chunked_message) == 0:
+            self.close()
+            return b''
+        else:
+            #  The first 4 bytes gives the total_byte_length and then
+            #  complete message is extracted
+            while total_byte_parsed < read_buffer:
+                # Case 1 - If the total_byte_length is partially read
+                # in the chunked message , then complete the total_byte_length
+                # by reading the required bytes from response and then
+                # generate the complete message
+                if read_buffer - total_byte_parsed <= 4:
+                    value = chunked_message[total_byte_parsed:
+                                            total_byte_parsed +
+                                            (read_buffer - total_byte_parsed) +
+                                            1]
+                    rem_bytes = self.response.read(4 - (read_buffer -
+                                                   total_byte_parsed)
+                                              )
+                    message = value + rem_bytes + \
+                        self.response.read(byte_int(value+rem_bytes)-4)
+                    end_status = self.decode_message(message, rec)
+                    total_byte_parsed = 0
+                    break
+                else:
+                    total_byte_length = chunked_message[total_byte_parsed:
+                                                        total_byte_parsed + 4]
+                    # Case 2 - Incomplete message in chunked message ,
+                    # so creating the complete message by reading the
+                    # total_byte_length- len_read from the response message.
+                    if total_byte_parsed + byte_int(total_byte_length) > read_buffer:
+                        len_read = len(chunked_message[total_byte_parsed:])
+                        message = chunked_message[total_byte_parsed:] + \
+                            self.response.read(byte_int(total_byte_length)-len_read)
+                        end_status = self.decode_message(message, rec)
+                        total_byte_parsed += byte_int(total_byte_length)
+                    # Case 3- the complete message is present in chunked
+                    # messsage.
+                    else:
+                        message = chunked_message[total_byte_parsed:
+                                                  total_byte_parsed +
+                                                  byte_int(total_byte_length)]
+                        total_byte_parsed += byte_int(total_byte_length)
+                        end_status = self.decode_message(message, rec)
+                if end_status:
+                    break
+            return rec
+
+    def extract_header(self, header, header_length):
```

Comment:
All the private functions such as these should have `_` prefix - while they are no explicit private functions. It is customary to use `_` to indicate private functions in python.

Reply:
done